akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -335,6 +335,70 @@ class ConnectionPoolSpec extends AkkaSpec("""
     }
   }
 
+  "be able to handle 500 `Connection: close` requests against the test server" in new TestSetup {
+    val settings = ConnectionPoolSettings(system).withMaxConnections(4)
+    val poolFlow = Http().cachedHostConnectionPool[Int](serverHostName, serverPort, settings = settings)
+
+    val N = 500
+    val requestIds = Source.fromIterator(() ⇒ Iterator.from(1)).take(N)
+    val idSum = requestIds.map(id ⇒ HttpRequest(uri = s"/r$id").withHeaders(Connection("close")) → id).via(poolFlow).map {
+      case (Success(response), id) ⇒
+        requestUri(response) should endWith(s"/r$id")
+        id
+      case x ⇒ fail(x.toString)
+    }.runFold(0)(_ + _)
+
+    (1 to N).foreach(_ ⇒ acceptIncomingConnection())
+
+    Await.result(idSum, 10.seconds) shouldEqual N * (N + 1) / 2
+  }
+
+  "xoxo be able to handle 500 pipelined requests with connection termination" in new TestSetup(autoAccept = true) {
```

Comment:
you can remove the `xoxo` btw ;) That's only a trick I use to `testOnly *Clazz* -- -z xoxo` to run a specific test